src/modules/quoiFeur/quoiFeur.helpers.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  ChannelType,
+  type ChatInputCommandInteraction,
+  Guild,
+  type Message,
+  Role,
+} from 'discord.js';
+
+import { cache } from '../../core/cache';
+import { endWithQuoi } from '../../helpers/regex.helper';
+
+const ONE_MINUTE = 1 * 60 * 1000;
+const MUTED_BY_BOT = 'Muted by bot';
+
+const reactWithFeur = async (message: Message) => {
+  await message.react('🇫');
+  await message.react('🇪');
+  await message.react('🇺');
+  await message.react('🇷');
+};
+
+const reactWithCoubeh = async (message: Message) => {
+  await message.react('🇨');
+  await message.react('🇴');
+  await message.react('🇺');
+  await message.react('🇧');
+  await message.react('🇪');
+  await message.react('🇭');
+  await message.react('🔇');
+
+  const mutedRole = message.guild?.roles.cache.find((r) => r.name === MUTED_BY_BOT);
+
+  if (!mutedRole?.id) return;
+
+  await message.member?.roles.add(mutedRole.id);
+
+  setTimeout(() => {
+    message.member?.roles.remove(mutedRole.id).catch(console.error);
+  }, ONE_MINUTE * 5);
+};
+
+export const reactOnEndWithQuoi = async (message: Message) => {
+  const channelIds = await cache.get('quoiFeurChannels', []);
+  const channelHasGame = channelIds.find((channelId) => channelId === message.channelId);
+  if (!channelHasGame) return;
+
+  if (!endWithQuoi(message.content)) return;
+
+  const probability = 1 / 20;
+
+  try {
+    Math.random() <= probability ? await reactWithCoubeh(message) : await reactWithFeur(message);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const createRoleMutedByBot = async (guild: Guild | null): Promise<Role> => {
+  if (!guild) {
+    throw new Error('Guild is null in createRoleMutedByBot');
+  }
+  const existingMutedByBot = guild.roles.cache.find((role) => role.name === MUTED_BY_BOT);
+
+  return (
+    existingMutedByBot ??
+    guild.roles.create({
+      name: MUTED_BY_BOT,
+    })
+  );
+};
+
+export const deleteRoleMutedByBot = async (guild: Guild | null): Promise<void> => {
```

Comment:
This shouldn't depend on a specific guild but rather run for any guild the bot is in.

So I recommend to remove the role for every guild the bot is in.
(otherwise if you invite the same bot instance to another server it won't work)

I think you can change the signature of this and take a Client as input instead, so you can map it to the `ready` event directly.

(or even code it inline if you want)